demos/blog/blog.py

```diff
@@ -186,7 +186,7 @@ class ComposeHandler(BaseHandler):
             entry = await self.query("SELECT * FROM entries WHERE id = %s", int(id))
             if not entry:
                 raise tornado.web.HTTPError(404)
-            slug = entry.slug
+            slug = entry[0].slug
```

Comment:
it looks like there's `self.queryone()` for this purpose, but it raises `NoResultError` instead of returning `None` for no result, but other handlers use it with the expectation that it returns None instead of raising `NoResultError`, so there's a bunch of little bugs around this kind of thing ...

Reply:
Thanks. Using queryone() and surrounding it with a try..except block may be more appropriate.